### PR TITLE
feat: new version grouping

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

File: docs/topics/multiplatform/multiplatform-compatibility-guide.md [hunk 1]
```diff
@@ -28,308 +33,206 @@ with available Gradle, Xcode, and Android Gradle plugin versions:
...
 
-We've added this deprecation warning due to fundamental compatibility issues between Kotlin Multiplatform's project model
-and Gradle's Java ecosystem plugins. Gradle's Java ecosystem plugins currently don't take into account that other plugins may:
+This section covers incompatible changes that end their deprecation cycle and come into effect in Kotlin 2.0.0−%kotlinVersion%.
 
-* Also publish or compile for the JVM target in a different way than the Java ecosystem plugins.
-* Have two different JVM targets in the same project, such as JVM and Android.
-* Have a complex multiplatform project structure with potentially multiple non-JVM targets.
+<anchor name="android-target-rename"/>
+### Rename of android target to androidTarget
```
Comment (zamulla) [inline docs/topics/multiplatform/multiplatform-compatibility-guide.md:41]:
```suggestion
### Renaming android target to androidTarget
```

Writerside supports inline formatting in headers, and while we generally don't endorse that — could this be the lesser of two evils here (make it "`android` target to `androidTarget`")? Otherwise it's very hard to parse the sentence without reading the text below.

File: docs/topics/multiplatform/multiplatform-compatibility-guide.md [hunk 2]
```diff
@@ -28,308 +33,206 @@ with available Gradle, Xcode, and Android Gradle plugin versions:
...
 
-Unfortunately, Gradle doesn't currently provide any API to address these issues.
+**What's changed?**
 
-We previously used some workarounds in Kotlin Multiplatform to help with the integration of Java ecosystem plugins.
-However, these workarounds never truly solved the compatibility issues, and since the release of Gradle 8.8, these workarounds
-are no longer possible. For more information, see our [YouTrack issue](https://youtrack.jetbrains.com/issue/KT-66542/Gradle-JVM-target-with-withJava-produces-a-deprecation-warning).
+We continue our efforts to stabilize Kotlin Multiplatform. An essential step in this way is to provide first-class
```
Comment (zamulla) [inline docs/topics/multiplatform/multiplatform-compatibility-guide.md:45]:
```suggestion
We continue our efforts to stabilize Kotlin Multiplatform. An essential step in this direction is to provide first-class
```

To me "efforts to stabilize" means that KMP is not stable yet, which may be confusing, because it is. Maybe it's more precise to say "to make it more stable"?

File: docs/topics/multiplatform/multiplatform-compatibility-guide.md [hunk 3]
```diff
@@ -28,308 +33,206 @@ with available Gradle, Xcode, and Android Gradle plugin versions:
...
+support for the Android target. In the future, this support will be provided via a separate plugin, developed by the
+Android team from Google.
 
-While we don't yet know exactly how to resolve this compatibility problem, we are committed to continuing support for
-some form of Java source compilation in your Kotlin Multiplatform projects. At a minimum, we will support the compilation
-of Java sources and using Gradle's [`java-base`](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/JavaBasePlugin.html) plugin within your multiplatform projects.
+To open the way for the new solution from Google, we're renaming the `android` block to `androidTarget` in the current
```
Comment (zamulla) [inline docs/topics/multiplatform/multiplatform-compatibility-guide.md:49]:
```suggestion
To open the way for the new solution, we're renaming the `android` block to `androidTarget` in the current
```

Super nitpicky, it just felt like too much Google for two short paragraphs.

File: docs/topics/multiplatform/multiplatform-compatibility-guide.md [hunk 4]
```diff
@@ -28,308 +33,206 @@ with available Gradle, Xcode, and Android Gradle plugin versions:
...
 ```
 
-</tab>
-</tabs>
-
+The implementation comes with non-trivial configuration complexity:
 
-Your parent project is now set up to work with both plugins.
+* You have to set up Gradle attributes on the `:shared` side and each consumer's side. Otherwise, Gradle can't
+  resolve dependencies in such projects because without additional information  it's not clear whether the consumer
```
Comment (zamulla) [inline docs/topics/multiplatform/multiplatform-compatibility-guide.md:122]:
```suggestion
  resolve dependencies in such projects because without additional information it's not clear whether the consumer
```

**Review by zamulla — APPROVED** [review]
Way easier to navigate!

File: docs/topics/multiplatform/multiplatform-compatibility-guide.md (hunk 1 above)
Comment (danil-pavlov) [reply docs/topics/multiplatform/multiplatform-compatibility-guide.md]: This was definitely a limitation from webhelp, now we can use proper syntax in headers
